common/statistics/src/report.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::clients::{
+    gateway_conn_statistics::GatewayStats, nym_api_statistics::NymApiStats,
+    packet_statistics::PacketStatistics,
+};
+
+use super::error::StatsError;
+
+use serde::{Deserialize, Serialize};
+use time::OffsetDateTime;
+
+/// Report object containing both data to be reported and client / device context. We take extra care not to overcapture context information.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ClientStatsReport {
+    pub(crate) last_update_time: OffsetDateTime,
+    pub(crate) client_id: String,
+    pub(crate) client_type: String,
+    pub(crate) os_information: os_info::Info,
+    pub(crate) packet_stats: PacketStatistics,
+    pub(crate) gateway_conn_stats: GatewayStats,
+    pub(crate) nym_api_stats: NymApiStats,
+}
+
+impl TryFrom<ClientStatsReport> for Vec<u8> {
+    type Error = StatsError;
+
+    fn try_from(value: ClientStatsReport) -> Result<Self, Self::Error> {
+        let report_json = serde_json::to_string(&value)?;
+        Ok(report_json.as_bytes().to_vec())
+    }
+}
+
+impl TryFrom<Vec<u8>> for ClientStatsReport {
```

Comment:
nitpicking: usually, for bytes conversion the associated methods take `&[u8]` as arguments since the bytes are hardly ever consumed and it makes them more versatile

Reply:
Fixed in https://github.com/nymtech/nym/pull/5107/commits/0b0f2c77cad0831e84bf88efe3dc94dcf9931faa